core/src/main/java/com/linecorp/armeria/internal/common/stream/FixedStreamMessage.java

```diff
@@ -112,12 +111,19 @@ public void subscribe(Subscriber<? super T> subscriber, EventExecutor executor,
         requireNonNull(executor, "executor");
         requireNonNull(options, "options");
         if (isOpen()) {
-            abortSubscriber(executor, subscriber, "a fixed stream is not closed yet");
+            abortSubscriber(executor, subscriber,
+                            new IllegalStateException("a fixed stream is not closed yet"));
             return;
         }
 
-        if (!subscribedUpdater.compareAndSet(this, 0, 1)) {
-            abortSubscriber(executor, subscriber, "subscribed by other subscriber already");
+        if (!executorUpdater.compareAndSet(this, null, executor)) {
+            final Throwable abortCause = this.abortCause;
+            if (abortCause == null) {
+                abortSubscriber(executor, subscriber,
+                                new IllegalStateException("subscribed by other subscriber already"));
+            } else {
+                abortSubscriber(executor, subscriber, abortCause);
```

Comment:
Question) I'm not sure if I follow the reasoning for this behavior change.

Now, it seems like if `abort` and `subscribe` is called in quick succession, the aborted cause will be returned.

Would it be better to just always return a `"subscribed by other subscriber already"` for double subscription so that users are more aware of what went wrong?

Reply:
I'm not sure we can assume abortion is a subscription. Because users did not set a subscriber to the stream. `abortCause` could tell an exception is closed exceptionally and a subscription is no longer available.

Reply:
> I'm not sure we can assume abortion is a subscription

Good point, I didn't think of it this way.
Also checked that `CancellableStreamMessage` has the same logic 👍